databases/business-sync/fragment-shoebox-snippet.js

```diff
@@ -0,0 +1,40 @@
+{
+  channels: getDocSyncChannels(doc, oldDoc),
+  typeFilter: function(doc, oldDoc) {
+    return createBusinessEntityRegex('shoeboxSnippet\\.[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]+').test(doc._id);
+  },
+  allowUnknownProperties: true,
+  immutable: true,
+  propertyValidators: {
+    type: {
+      // The type of shoebox snippet (ie, bank transaction, email, photo, digital receipt)
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
```

Comment:
Anytime I come across a `type` field it feels like it scream enum to me so I agree with Ziemek's hunch. I know with synctos it will be easier to have the type safety check. I think the question comes down to how challenging they are to evolve and refactor. Probably not a lot different than in Java where it is doable but not something you undertake lightly (although the implied backing by a sequence of integers probably doesn't exist and makes it easier).